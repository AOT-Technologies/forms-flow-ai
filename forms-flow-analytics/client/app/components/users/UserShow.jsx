import React from "react";
import { includes } from "lodash";
import Tag from "antd/lib/tag";
import Group from "@/services/group";
import { UserProfile } from "../proptypes";

export default class UserShow extends React.Component {
  static propTypes = {
    user: UserProfile.isRequired,
  };

  constructor(props) {
    super(props);
    this.state = { groups: [], loadingGroups: true };
  }

  componentDidMount() {
    Group.query().then(groups => {
      this.setState({ groups, loadingGroups: false });
    });
  }

  renderUserGroups() {
    const { groupIds } = this.props.user;
    const { groups } = this.state;

    return (
      <div>
        {groups
          .filter(group => includes(groupIds, group.id))
          .map(group => (
            <Tag className="m-t-5 m-r-5" key={group.id}>
              <a href={`groups/${group.id}`}>{group.name}</a>
            </Tag>
          ))}
      </div>
    );
  }

  render() {
    const { name, email, profileImageUrl } = this.props.user;
    const { loadingGroups } = this.state;

    return (
      <div className="col-md-4 col-md-offset-4 profile__container">
        <img alt="profile" src={profileImageUrl} className="profile__image" width="40" />

        <h3 className="profile__h3">{name}</h3>

        <hr />

        <dl className="profile__dl">
          <dt>Name:</dt>
          <dd>{name}</dd>
          <dt>Email:</dt>
          <dd>{email}</dd>
          <dt>Groups:</dt>
          <dd>{loadingGroups ? "Loading..." : this.renderUserGroups()}</dd>
        </dl>
      </div>
    );
  }
}
