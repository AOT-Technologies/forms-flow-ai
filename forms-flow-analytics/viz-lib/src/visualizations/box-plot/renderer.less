.box-plot-deprecated-visualization-container {
  overflow: hidden;
  height: 500px;
}
