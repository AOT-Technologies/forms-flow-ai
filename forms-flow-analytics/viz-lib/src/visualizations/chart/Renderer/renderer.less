.chart-visualization-container {
  height: 400px;
  overflow: hidden;
}
