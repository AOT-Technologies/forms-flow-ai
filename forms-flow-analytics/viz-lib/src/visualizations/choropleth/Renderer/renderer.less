.choropleth-visualization-legend {
  padding: 3px;
  cursor: default;

  > div {
    line-height: 1;
    margin: 5px;
  }

  .legend-item {
    display: flex;
    align-items: center;

    .color-swatch {
      margin-right: 5px;
    }

    .legend-item-text {
      flex: 1 1 auto;

      &.text-left {
        text-align: left;
      }
      &.text-center {
        text-align: center;
      }
      &.text-right {
        text-align: right;
      }
    }
  }
}
