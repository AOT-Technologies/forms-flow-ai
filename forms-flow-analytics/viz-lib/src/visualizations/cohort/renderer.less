@import "../variables";

.cohort-visualization-container {
  .cornelius-table {
    width: 100%;

    &,
    tr,
    th,
    td {
      border-color: #f0f0f0;
    }

    .cornelius-time,
    .cornelius-label,
    .cornelius-stage,
    .cornelius-people {
      background-color: fade(@visualizations-gray, 3%);
    }
  }
}
