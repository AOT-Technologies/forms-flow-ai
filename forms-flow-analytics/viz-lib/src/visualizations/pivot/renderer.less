@import "../variables.less";

.pivot-table-visualization-container {
  &[data-hide-controls] {
    .pvtAxisContainer,
    .pvtRenderers,
    .pvtVals {
      display: none;
    }
  }

  &[data-hide-row-totals] {
    td:last-child,
    th:last-child {
      &.pvtTotalLabel:not(:empty),
      &.pvtTotal,
      &.pvtGrandTotal {
        display: none;
      }
    }
  }

  &[data-hide-column-totals] {
    tbody > tr:last-child {
      & > .pvtTotalLabel,
      & > .pvtTotal,
      & > .pvtGrandTotal {
        display: none;
      }
    }
  }
}

.pvtAxisContainer,
.pvtVals {
  border: 1px solid fade(@visualizations-gray, 15%);
  background: #fff;
}

.pvtAxisContainer li span.pvtAttr {
  background: fade(@visualizations-gray, 10%);
  border: 1px solid fade(@visualizations-gray, 15%);
  border-radius: 3px;
}

.pvtCheckContainer {
  border-top: 1px solid fade(@visualizations-gray, 15%);
  border-bottom: 1px solid fade(@visualizations-gray, 15%);
}

.pvtCheckContainer p {
  margin: 2px;
  line-height: 1;
}

.pvtTriangle {
  color: fade(@visualizations-gray, 90%);
}

table.pvtTable thead tr th,
table.pvtTable tbody tr th {
  background-color: fade(@visualizations-gray, 10%);
  border: 1px solid #ced8dc;
}

table.pvtTable tbody tr td {
  border: 1px solid #ced8dc;
}

.pvtFilterBox {
  border: 1px solid fade(@visualizations-gray, 15%);
  border-radius: 3px;
  box-shadow: fade(@visualizations-gray, 15%) 0px 4px 9px -3px;

  button {
    background-color: rgba(102, 136, 153, 0.15);
    margin-right: 5px;
    border: 1px solid transparent;
    padding: 3px 6px;
    font-size: 13px;
    line-height: 1.42857143;
    border-radius: 3px;

    &:hover {
      background-color: rgba(102, 136, 153, 0.25);
    }
  }

  input[type="text"] {
    width: 90%;
    margin: 0 auto 10px;
    height: 35px;
    padding: 6px 12px;
    border: 1px solid #e8e8e8;
    border-radius: 3px;
  }
}
