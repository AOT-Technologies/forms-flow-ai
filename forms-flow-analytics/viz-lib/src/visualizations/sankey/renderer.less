/* Sankey Visualization */
.sankey .node rect {
  fill-opacity: .9;
  shape-rendering: crispEdges;
  stroke-width: 0;
}
.sankey .node text {
  text-shadow: 0 1px 0 #fff;
}
.sankey .link {
  fill: none;
  stroke: #000;
  stroke-opacity: .2;
}

.sankey-visualization-container {
  height: 500px;
  overflow: hidden;
}
