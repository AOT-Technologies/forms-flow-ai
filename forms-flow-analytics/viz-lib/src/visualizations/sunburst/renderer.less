.sunburst-visualization-container {
  height: 400px;
  display: flex;
  flex-direction: column;

  > div {
    position: relative;

    &:first-child {
      flex-grow: 0;
    }
    &:last-child {
      flex-grow: 1;
    }
  }

  .sunburst-container,
  .summary-container {
    position: absolute;
    left: 0;
    top: 0;
    right: 0;
    bottom: 0;
    width: auto;
    height: auto;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .summary-container {
    font-size: 11px;
    color: #666;
  }
}
