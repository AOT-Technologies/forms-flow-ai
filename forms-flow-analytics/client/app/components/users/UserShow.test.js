import React from "react";
import renderer from "react-test-renderer";
import Group from "@/services/group";
import UserShow from "./UserShow";

beforeEach(() => {
  Group.query = jest.fn().mockResolvedValue([]);
});

test("renders correctly", () => {
  const user = {
    id: 2,
    name: "John Doe",
    email: "john@doe.com",
    groupIds: [],
    profileImageUrl: "http://www.images.com/llama.jpg",
  };

  const component = renderer.create(<UserShow user={user} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
