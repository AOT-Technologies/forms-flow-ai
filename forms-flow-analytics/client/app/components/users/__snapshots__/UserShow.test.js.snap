// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="col-md-4 col-md-offset-4 profile__container"
>
  <img
    alt="profile"
    className="profile__image"
    src="http://www.images.com/llama.jpg"
    width="40"
  />
  <h3
    className="profile__h3"
  >
    John Doe
  </h3>
  <hr />
  <dl
    className="profile__dl"
  >
    <dt>
      Name:
    </dt>
    <dd>
      John Doe
    </dd>
    <dt>
      Email:
    </dt>
    <dd>
      john@doe.com
    </dd>
    <dt>
      Groups:
    </dt>
    <dd>
      Loading...
    </dd>
  </dl>
</div>
`;
